src/platform/SDL2Window.cxx

```diff
@@ -187,128 +181,110 @@ void SDLWindow::swapBuffers()
 bool SDLWindow::create(void)
 {
     int targetWidth, targetHeight;
-    getSize(targetWidth, targetHeight);
-    SDL_bool windowWasGrabbed = SDL_FALSE;
-    if (windowId != NULL)
-        windowWasGrabbed = SDL_GetWindowGrab(windowId);
-
-    // if we have an existing identical window, go no further
-    if (windowId != NULL)
+    // If fullscreen, the target size will be that of the selected fullscreen resolution
+    if (fullScreen)
+        const_cast<SDLDisplay *>((const SDLDisplay *)getDisplay())->getWindowSize(targetWidth, targetHeight);
+    // Otherwise, it will be the base size, which controls the windowed size
+    else
     {
-        int currentWidth, currentHeight;
-        SDL_GetWindowSize(windowId, &currentWidth, &currentHeight);
-
-        Uint32 priorWindowFlags = SDL_GetWindowFlags(windowId);
-        if (fullScreen == (priorWindowFlags & SDL_WINDOW_FULLSCREEN) &&
-                targetWidth == currentWidth && targetHeight == currentHeight)
-            return true;
+        targetWidth = base_width;
+        targetHeight = base_height;
     }
 
-    // destroy the pre-existing window if it exists
-    if (windowId != NULL)
+    // Create the SDL window if it doesn't already exist
+    if (!windowId)
     {
-        if (glContext)
-            SDL_GL_DeleteContext(glContext);
-        glContext = NULL;
-
-        SDL_DestroyWindow(windowId);
-    }
-
-    // (re)create the window
-    const Uint32 flags = SDL_WINDOW_OPENGL |
-                         (fullScreen ? SDL_WINDOW_FULLSCREEN : SDL_WINDOW_RESIZABLE) |
-                         (windowWasGrabbed ? SDL_WINDOW_INPUT_GRABBED : 0);
+        const Uint32 flags = SDL_WINDOW_OPENGL | SDL_WINDOW_RESIZABLE;
+        windowId = SDL_CreateWindow(
+                       title.c_str(),
+                       SDL_WINDOWPOS_UNDEFINED,
+                       SDL_WINDOWPOS_UNDEFINED,
+                       base_width,
+                       base_height,
+                       flags);
+
+        // If the window could not be created, bail out
+        if (!windowId)
+        {
+            printf("Could not create the window: %s\n", SDL_GetError());
+            return false;
```

Comment:
what happens in this case? There is no window, can the game be played? Multiple returns are bad. 

I wonder if this might be one of those exceptional conditions where throwing an exception is warranted?

Reply:
// set fullscreen again so MainWindow object knows it's full screen
    if (useFullscreen)
        // this will also call window create
        pmainWindow->setFullscreen();
    else
        window->create();
The return value is ignored.

It looks like he was just following the existing pattern. We do need a way to bail out from errors like this, but since that involves cleanup inside SDL (e.g.,` SDL_Quit()`) and possibly outside of SDL as well, that seems more like a separate project entirely. SDL has functions to display an error dialog, or if SDL initialization itself failed, we can always fall back to stderr output. Yet another thing that would be simplified by going all in on SDL.